system/Validation/Rules.php

```diff
@@ -353,4 +353,22 @@ public function required_without(
 
         return true;
     }
+
+    /**
+     * The field exists in $data.
+     *
+     * @param array|bool|float|int|object|string|null $value The field value.
+     * @param string|null                             $param The rule's parameter.
+     * @param array                                   $data  The data to be validated.
+     * @param string|null                             $field The field name.
+     */
+    public function field_exists(
+        $value = null,
+        ?string $param = null,
+        array $data = [],
+        ?string $error = null,
+        ?string $field = null
+    ): bool {
+        return array_key_exists($field, $data);
```

Comment:
Ah, we need to handle the key with dot array syntax...

Reply:
Good call, I'm afraid so. See `required_without` above. May be worth having a private method for `$data` transformations.

Reply:
Ah, I see your next PR - much better solution.

Reply:
Yes, done.